templates/haproxy.cfg.j2

```diff
@@ -53,27 +53,46 @@ listen stats
     stats realm Haproxy\ Statistics
     stats uri {{ haproxy_stats_uri }}
 {% endif %}
-frontend {{ haproxy_frontend_name }}
-    bind {{ haproxy_frontend_bind_address }}:{{ haproxy_frontend_port }}
-    mode {{ haproxy_frontend_mode }}
-    default_backend {{ haproxy_backend_name }}
 
-backend {{ haproxy_backend_name }}
-    mode {{ haproxy_backend_mode }}
-    balance {{ haproxy_backend_balance_method }}
+{% for frontend in haproxy_frontends %}
+frontend {{ frontend['name'] }}
+    bind {{ frontend['bind_address'] }}:{{ frontend['port'] }}
+    mode {{ frontend['mode'] }}
+{% if frontend.has_key('acls') and frontend.has_key('conditions') %}
```

Comment:
dict.has_key() is removed in future versions of Python, use the in operator instead

    {% if 'acls' in frontend and 'conditions' in frontend %}